packages/eslint-plugin/src/rules/thenable-in-promise-aggregators.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  isTypeAnyType,
+  isTypeUnknownType,
+} from '@typescript-eslint/type-utils';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+
+import { createRule, getParserServices } from '../util';
+
+export default createRule({
+  name: 'thenable-in-promise-aggregators',
+  meta: {
+    docs: {
+      description:
+        'Disallow passing non-Thenable values to promise aggregators',
+      recommended: 'recommended',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      inArray:
+        'Unexpected non-Thenable value in array passed to promise aggregator.',
+      arrayArg:
+        'Unexpected array of non-Thenable values passed to promise aggregator.',
+      nonArrayArg: 'Unexpected non-array passed to promise aggregator.',
+    },
+    schema: [],
+    type: 'problem',
+  },
+  defaultOptions: [],
+
+  create(context) {
+    const services = getParserServices(context);
+    const checker = services.program.getTypeChecker();
+
+    return {
+      CallExpression(node: TSESTree.CallExpression): void {
+        if (node.callee.type !== AST_NODE_TYPES.MemberExpression) {
+          return;
+        }
```

Comment:
We should skip `ChainExpression`s here because
```ts
(Promise?.all)([1])
```

Won't be checked, but

```ts
Promise?.all([1])
```
will be

Reply:
Btw, this logic is very similar to the logic from https://github.com/typescript-eslint/typescript-eslint/pull/8011

https://github.com/typescript-eslint/typescript-eslint/pull/8011/files#diff-801508874bd1d1cb9f122a9ca577fdf2350bfd756c5e63572487b65b859d6799R91-R102

Maybe we can reuse it between these two rules, what do you think?

Reply:
(Because of hidden comments I saw this in the wrong order)

It is very similar; I think they can be reused! I wrote it in a way so that it should be easy to adapt once one gets merged. Let me know if you'd rather break it out before one gets merged (and I guess rebase on the other one?)

Reply:
I moved both `isPromiseLike` and `isPromiseConstructorLike` to the `type-utils` in https://github.com/typescript-eslint/typescript-eslint/pull/8011/commits/a0699cfa5c5143a38c38f8d746a28688d736fd74#diff-c7675022367e4f5ec62101a590d4f77765532a6ec1a4e6e628b407a082bcf2fd, so we can easily reuse them later. Not sure about `skipChainExpression` though